back/app/models/cosponsors_initiative.rb

```diff
@@ -27,6 +27,19 @@ class CosponsorsInitiative < ApplicationRecord
   belongs_to :user
   belongs_to :initiative
 
+  before_destroy :remove_notifications # Must occur before has_many :notifications (see https://github.com/rails/rails/issues/5205)
+  has_many :notifications, dependent: :nullify
+
   validates :user, :initiative, presence: true
   validates :status, inclusion: { in: STATUSES }
+
+  private
+
+  def remove_notifications
+    notifications.each do |notification|
+      unless notification.update cosponsors_initiative: nil
```

Comment:
In what case it can return false?

Reply:
We agreed this looks fine (this pattern is used in all models associated with `notifications`)